segmentation/utils/io_utils.py

```diff
@@ -1,8 +1,123 @@
 import os
+import warnings
 
 
-def list_folders(input_dir):
-    files = os.listdir(input_dir)
-    folders = [file for file in files if os.path.isdir(os.path.join(input_dir, file))]
+def list_files(dir_name, substrs=None):
+    """ List all files in a directory containing at least one given substring
 
-    return folders
+    Args:
+        dir_name (str):         Parent directory for files of interest
+        substrs (str or list):  Substring matching criteria, defaults to None (all files)
+
+    Returns:
+        matches (list):         List of files containing at least one of the substrings
+
+    """
+
+    files = os.listdir(dir_name)
+    files = [file for file in files if not os.path.isdir(os.path.join(dir_name, file))]
+
+    # default to return all files
+    if substrs is None:
+        return files
+
+    # handle case where substrs is a single string (not wrapped in list)
+    if type(substrs) is not list:
+        substrs = [substrs]
+
+    matches = []
+    for substr in substrs:
+        matches.extend([
+            file
+            for file in files
+            if substr in file
+        ])
```

Comment:
I was considering using regex, but I think just using `in` is probably easier and faster for now.  Regardless, I was being silly since this can just be:
```
matches = [file for file in files if file in substrs]
```
_O_(s\*f^2) -> _O_(s\*f) lol